test/integration2/schema/foreign-keys.spec.js

```diff
@@ -221,6 +220,25 @@ describe('Schema', () => {
             );
             expect(fks.length).to.equal(1);
           });
+
+          it('can add a new column with a foreign key constraint', async () => {
+            if (!isSQLite(knex)) {
+              return;
+            }
+
+            await knex.schema.alterTable('foreign_keys_table_one', (table) => {
+              table
+                .integer('fkey_new')
+                .unsigned()
+                .notNull()
+                .references('foreign_keys_table_two.id');
+            });
+
+            const fks = await knex.raw(
```

Comment:
why not make this test dialect-indepentent by triggering constraint violation?

Reply:
The problem is that SQLite is the only dialect that uses `statementsProducer()`. When adding a column and adding a foreign key constraint at the same time, the statements producer to add the constaint will simply not execute. Thus it is not possible to trigger a constraint violation, as there is no constraint to be violated. That's why the test is checking for the presence of the constraint.